service/defaultcomponents/default_exporters_test.go

```diff
@@ -0,0 +1,215 @@
+			},
+		},
+		{
+			exporter: "logging",
+		},
+		{
+			exporter: "opencensus",
+			getConfigFn: func() configmodels.Exporter {
+				cfg := expFactories["opencensus"].CreateDefaultConfig().(*opencensusexporter.Config)
+				cfg.GRPCClientSettings = configgrpc.GRPCClientSettings{
+					Endpoint: endpoint,
+				}
+				return cfg
+			},
+		},
+		{
+			exporter: "otlp",
+			getConfigFn: func() configmodels.Exporter {
+				cfg := expFactories["otlp"].CreateDefaultConfig().(*otlpexporter.Config)
+				cfg.GRPCClientSettings = configgrpc.GRPCClientSettings{
+					Endpoint: endpoint,
+				}
+				return cfg
+			},
+		},
+		{
+			exporter: "otlphttp",
+			getConfigFn: func() configmodels.Exporter {
+				cfg := expFactories["otlphttp"].CreateDefaultConfig().(*otlphttpexporter.Config)
+				cfg.Endpoint = "http://" + endpoint
+				return cfg
+			},
+		},
+		{
+			exporter: "prometheus",
+			getConfigFn: func() configmodels.Exporter {
+				cfg := expFactories["prometheus"].CreateDefaultConfig().(*prometheusexporter.Config)
+				cfg.Endpoint = endpoint
+				return cfg
+			},
+		},
+		{
+			exporter: "prometheusremotewrite",
+		},
+		{
+			exporter: "zipkin",
+			getConfigFn: func() configmodels.Exporter {
+				cfg := expFactories["zipkin"].CreateDefaultConfig().(*zipkinexporter.Config)
+				cfg.Endpoint = endpoint
+				return cfg
+			},
+		},
+	}
+
+	assert.Equal(t, len(tests), len(expFactories))
+	for _, tt := range tests {
+		t.Run(string(tt.exporter), func(t *testing.T) {
+			factory, ok := expFactories[tt.exporter]
+			require.True(t, ok)
+			assert.Equal(t, tt.exporter, factory.Type())
+			assert.Equal(t, tt.exporter, factory.CreateDefaultConfig().Type())
+
+			verifyExporterLifecycle(t, factory, tt.getConfigFn)
+		})
+	}
+}
+
+// GetExporterConfigFn is used customize the configuration passed to the verification.
+// This is used to change ports or provide values required but not provided by the
+// default configuration.
+type getExporterConfigFn func() configmodels.Exporter
+
+// verifyExporterLifecycle is used to test if an exporter type can handle the typical
+// lifecycle of a component. The getConfigFn parameter only need to be specified if
+// the test can't be done with the default configuration for the component.
+func verifyExporterLifecycle(t *testing.T, factory component.ExporterFactory, getConfigFn getExporterConfigFn) {
+	ctx := context.Background()
+	host := newAssertNoErrorHost(t)
+	expCreateParams := component.ExporterCreateParams{
+		Logger:               zap.NewNop(),
+		ApplicationStartInfo: component.DefaultApplicationStartInfo(),
+	}
+
+	if getConfigFn == nil {
+		getConfigFn = factory.CreateDefaultConfig
+	}
+
+	createFns := []createExporterFn{
+		wrapCreateLogsExp(factory),
+		wrapCreateTracesExp(factory),
+		wrapCreateMetricsExp(factory),
+	}
+
+	for _, createFn := range createFns {
+		firstExp, err := createFn(ctx, expCreateParams, getConfigFn())
+		if err == configerror.ErrDataTypeIsNotSupported {
+			continue
+		}
```

Comment:
`if errors.Is(err, configerror.ErrDataTypeIsNotSupported)`

Reply:
done